omnipaxos_storage/src/memory.rs

```diff
@@ -0,0 +1,374 @@
+#[allow(missing_docs)]
+pub mod persistent_storage {
+    use commitlog::{
+        message::{MessageBuf, MessageSet},
+        CommitLog, LogOptions, ReadLimit,
+    };
+    use omnipaxos_core::{
+        ballot_leader_election::Ballot,
+        storage::{Entry, Snapshot, StopSignEntry, Storage},
+    };
+    use rocksdb::{Options, DB};
+    use std::marker::PhantomData;
+    use zerocopy::{AsBytes, FromBytes};
+    const COMMITLOG: &str = "commitlog/";
+    const ROCKSDB: &str = "rocksDB/";
+
+    /// A persistent storage implementation, lets sequence paxos write the log
+    /// and variables (e.g. promised, accepted) to disk. Log entries are serialized
+    /// and de-serialized into slice of bytes when read or written from the log.
+    //#[derive(Debug)]
+    pub struct PersistentState<T, S>
+    where
+        T: Entry,
+        S: Snapshot<T>,
+    {
+        /// a disk-based commit log for entries
+        c_log: CommitLog,
+        /// path to commitlog, remove when commitlog is no longer deleted here
+        c_log_path: String,
+        /// a struct for accessing local RocksDB database
+        db: DB,
+        /// Garbage collected index.
+        trimmed_idx: u64,
+        /// Stored snapshot
+        snapshot: Option<S>,
+        /// Stored StopSign
+        stopsign: Option<StopSignEntry>,
```

Comment:
These are still in memory... not sure if you can store them in rocks but they need to be persisted too.